PurpleExplorer/Views/MainWindow.xaml.cs

```diff
@@ -32,11 +32,17 @@ private void TreeView_SelectionChanged(object sender, SelectionChangedEventArgs
             var mainWindowViewModel = DataContext as MainWindowViewModel;
             var treeView = sender as TreeView;
 
-            mainWindowViewModel.ClearSelectedSubscription();
+            mainWindowViewModel.ClearSelection();
             if (treeView.SelectedItem is ServiceBusSubscription)
             {
-                var selectedItem = treeView.SelectedItem as ServiceBusSubscription;
-                mainWindowViewModel.SetSelectedSubscription(selectedItem);
+                var selectedSubscription = treeView.SelectedItem as ServiceBusSubscription;
+                mainWindowViewModel.SetSelectedSubscription(selectedSubscription);
+            }
+
+            if (treeView.SelectedItem is ServiceBusTopic)
+            {
+                var selectedTopic = treeView.SelectedItem as ServiceBusTopic;
+                mainWindowViewModel.SetSelectedTopic(selectedTopic);
```

Comment:
```suggestion
            if (treeView.SelectedItem is ServiceBusTopic selectedTopic)
            {
                mainWindowViewModel.SetSelectedTopic(selectedTopic);
```

I understand you were following the style of the previous block, but using that feature remove quite a bit of code.

Reply:
You could even use type-based pattern matching in a `switch` statement if you want, <https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/switch#type-pattern>

```csharp
switch (treeView.SelectedItem)
{
    case ServiceBusSubscription selectedSubscription:
        mainWindowViewModel.SetSelectedSubscription(selectedSubscription);
        break;

    case ServiceBusTopic selectedTopic:
        mainWindowViewModel.SetSelectedTopic(selectedTopic);
        break;
}
```

Reply:
Thanks Mick.  I didn't know about it.  Fixed it as you have suggested.